Call different Build and Test jobs based on BUILD_IDENTIFIER

AdamBrousseau opened the pull request:
- Keep Pipeine-Build-Test* jobs common
- Separates Build* and Test* jobs
- Set BuildDiscarder differently per BUILD_IDENTIFIER
   - Keep different number of build logs depending on
     which BUILD_IDENTIFIER the job is
   - Remove BuildDiscader for Artifacts since we are
     using Artifactory
- Change Spec names to match AdoptOpenJDK convention
- Change Build and Pipeline job names to more closely match AdoptOpenJDK
- Change Test job names to match AdoptOpenJDK
- Allow custom Git repo/branch for autogen jobs
- Remove DISCARDER_NUM_ARTIFACTS from autogen since we are
  using Artifactory now
- Add temporary reverse spec map to allow old specs from old
  style PR builds to map to the new spec names
- Remove SPEC from testing ENV so spec names don't conflict

[skip ci]
Related #2728 (comment)
Related #5210

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

AdamBrousseau commented:
Test Builds:
"Custom" (Personal)
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/25/
Nightly
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/26/
OMR-Acceptance
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/27/

AdamBrousseau commented:
@jdekonin for review

AdamBrousseau commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -69,13 +69,12 @@ test_dependencies_job_name: 'test.getDependency'
+    OMR: 20
+    OpenJDK: 20
+    Personal: 50
+    Pipeline: 100
```
@pshipton are these numbers reasonable to start with?

AdamBrousseau replied:
Note: OpenJDK would be for the OpenJDK Acceptance build if/when it is created.

jdekonin replied:
And Personal is for if/when personal build are created as well.  Those would come in handy for testing changes like this. :)

AdamBrousseau replied:
`Personal` is the default, so anything sandbox or custom and also PullRequests.

AdamBrousseau commented:
@samuel-rubin please review the DSL changes.

pshipton commented:
I'll wait until the artifactory issue is resolved or backed out, and also for the code split Monday. We can add into the 0.14 release branch if necessary afterwards.

AdamBrousseau commented:
Depends on #5210 as well.

Also I created all the new views with regexes to sort the new jobs when they are created.

llxia commented on buildenv/jenkins/common/pipeline-functions:
```diff
@@ -406,14 +406,29 @@ def get_test_target_names() {
-    return "Test-${targetName}-JDK${version}-${spec}"
+def get_test_job_name(targetName, spec, version, identifier) {
+    id = convert_build_identifier(identifier)
+    return "Test-${targetName}-JDK${version}-${spec}-${id}"
```
- Please see SPEC to ARCH_OS map in https://github.com/eclipse/openj9/issues/5210#issuecomment-478599394
- test job name format should be `Test_openjdk${version}_j9_${targetName}_${ARCH_OS}_${id}`

AdamBrousseau commented:
Updated based on new tst job names. Renamed specs to match Adopt convention.
First build here: jdk8 x86-64_linux /w sanity.functional
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/37/
Next build here: jdk8,11 xlinux, xlinuxXL, plinux
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/40/

@llxia @jdekonin @pshipton for review. I will need to coordinate an internal change with this. We still need to generate the rest of the test jobs via #5210. I will also need to send some emails to indicate advertise these changes.

AdamBrousseau commented:
Note, the existing pull request builds are all configured with the old spec names. They will be broken with this change as-is.
3 Options:
1. Merge #2836 as well. This will require a bit more testing and some merge conflict resolution (more delay and a larger change at once).
1. Manually update all ~65 PR jobs with the new spec names (boo)
1. Add a temp reverse map for the PR builds to convert the old spec names to the new spec names.
I think 3 is the best option.

AdamBrousseau commented:
Jenkins compile jdk8 plinux

AdamBrousseau commented:
Jenkins compile plinux jdk8

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk11

jdekonin commented on buildenv/jenkins/variables/defaults.yml:
```diff
@@ -258,7 +259,7 @@ linux_x86-64_cmprssptrs:
 #========================================#
-linux_x86-64_cmprssptrs_cmake:
+x86-64_linux_CM:
   boot_jdk:
```
make lower case for consistency.

AdamBrousseau commented:
Jenkins compile plinux jdk11

AdamBrousseau commented:
Jenkins compile plinux jdk8

AdamBrousseau commented:
jenkins test sanity plinux jdk11

AdamBrousseau commented:
The reverse spec map is working as expected. Just waiting on #5210 to be completed internally then this can be merged.

AdamBrousseau commented:
I take that back. Having SPEC set during the compile causes a conflict. Will investigate options.

AdamBrousseau commented:
Jenkins compile plinux jdk8

AdamBrousseau commented:
jenkins test sanity plinux jdk11

AdamBrousseau commented:
jenkins test sanity plinux jdk11

AdamBrousseau commented:
jenkins test sanity plinux jdk11

AdamBrousseau commented:
Removed the SPEC from the test env. Test compiles now. I think this is ready. I will draft an email for the dev mailing list. I think Lan is in progress for creating the internal jobs.
